### PR TITLE
[GEOS-11630] REST API throws HTTP 500 When Security Metadata Has Null Attributes

### PR DESCRIPTION
**Pull request by turingtestfail**
[GEOS-1630](https://osgeo-org.atlassian.net/browse/GEOS-1630) [Powered by Pull Request Badge](https://pullrequestbadge.com/?utm_medium=github&utm_source=geoserver&utm_campaign=badge_info)

  

# Checklist

- [x] I have read the [contribution guidelines](https://github.com/geoserver/geoserver/blob/main/CONTRIBUTING.md).
- [ ] I have sent a [Contribution Licence Agreement](https://docs.geoserver.org/latest/en/developer/policies/committing.html) (not required for small changes, e.g., fixing typos in documentation).
- [x] First PR targets the `main` branch (backports managed later; ignore for branch specific issues).
- [ ] All the build checks are green ([see automated QA checks](https://docs.geoserver.org/latest/en/developer/qa-guide/index.html)).

For core and extension modules:

- [x] New unit tests have been added covering the changes.
- [ ] [Documentation](https://github.com/geoserver/geoserver/tree/main/doc/en/user/source) has been updated (if change is visible to end users).
- [ ] The [REST API docs](https://github.com/geoserver/geoserver/tree/main/doc/en/api/1.0.0) have been updated (when changing configuration objects or the REST controllers).
- [x] There is an issue in the [GeoServer Jira](https://osgeo-org.atlassian.net/browse/GEOS/summary) (except for changes that do not affect administrators or end users in any way).
- [x] Commit message(s) must be in the form ``[GEOS-XYZWV] Title of the Jira ticket``.
- [x] Bug fixes and small new features are presented as a single commit.
- [x] Each commit has a single objective (if there are multiple commits, each has a separate JIRA ticket describing its goal).

### CODE REVIEW

File: src/main/src/main/java/org/geoserver/security/filter/GeoServerSecurityInterceptorFilter.java [hunk 1]
```diff
@@ -106,12 +108,14 @@ private AuthorizationDecision vote(
...
+        @SuppressWarnings("unchecked")
         @Override
         public AuthorizationDecision check(
                 Supplier<Authentication> authentication, HttpServletRequest request) {
-            Collection<ConfigAttribute> attributes = metadata.getAttributes(request);
-            AuthorizationDecision vote = vote(authentication.get(), request, attributes);
-            return vote;
+            Collection<ConfigAttribute> attributes =
+                    Optional.ofNullable(metadata.getAttributes(request))
+                            .orElse(Collections.EMPTY_SET);
```
[the remark this answers is not in the document]
Comment (turingtestfail) [reply src/main/src/main/java/org/geoserver/security/filter/GeoServerSecurityInterceptorFilter.java]: Done

File: src/main/src/main/java/org/geoserver/security/filter/GeoServerSecurityInterceptorFilter.java [hunk 2]
```diff
@@ -156,12 +160,14 @@ private AuthorizationDecision vote(
...
```
[the remark this answers is not in the document]
Comment (turingtestfail) [reply src/main/src/main/java/org/geoserver/security/filter/GeoServerSecurityInterceptorFilter.java]: Done

**Comment by sikeoka** [comment]
FYI, the actual ticket number is 11630 but this PR uses 1630

**Comment by turingtestfail** [comment]
> FYI, the actual ticket number is 11630 but this PR uses 1630

Thanks, fixed.

File: src/main/src/main/java/org/geoserver/security/RESTfulDefinitionSource.java [hunk 3]
```diff
@@ -29,10 +29,11 @@ public class RESTfulDefinitionSource implements FilterInvocationSecurityMetadata
 
     private static Logger log = Logging.getLogger(RESTfulDefinitionSource.class);
 
-    private static final String[] validMethodNames = {"GET", "PUT", "DELETE", "POST"};
+    private static final String[] validMethodNames = {"GET", "PUT", "DELETE", "POST", "HEAD"};
```
Comment (aaime) [inline src/main/src/main/java/org/geoserver/security/RESTfulDefinitionSource.java:32]: I was wondering about this change. Would you have gotten a 401 without it?
See also https://discourse.osgeo.org/t/http-options-request-fails-after-upgrade-to-2-26-1/111599/1

**Comment by turingtestfail** [comment]
Correct.  It would have returned a 401 on all HEAD requests.  We should add
OPTIONS to the list if that is supported by the REST API.

**Comment by turingtestfail** [comment]
Have you tried  curl -v -uadmin:geoserver -I
http://localhost:8080/geoserver/rest/about/status ?  There is a known issue
with using curl "-X HEAD" instead of "-I"

**Pull request merged**